packages/core/src/server.ts

```diff
@@ -1913,7 +1832,7 @@ export class Server {
     private _requestQueue = new Map<AccountID | OrgID, Promise<void>>();
 
     makeController(ctx: Context) {
-        return new Controller(this, ctx);
+        return new (V3Compat(Controller))(this, ctx);
```

Comment:
By using a mixin allows us to neatly separate the "compatibility layer" and the regular api. Once all v3 clients have disappeared from the wild, all we have to do is undo this change and delete the `v3-compat` module.